tools/quic/quic_transport_server.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+import argparse
+import asyncio
+import io
+import logging
+import os
+import re
+import struct
+import urllib.parse
+from typing import Dict, Optional
+
+from aioquic.asyncio import QuicConnectionProtocol, serve
+from aioquic.quic.configuration import QuicConfiguration
+from aioquic.quic.connection import END_STATES
+from aioquic.quic.events import StreamDataReceived, QuicEvent
+from aioquic.tls import SessionTicket
+
+SERVER_NAME = 'aioquic-transport'
+
+handlers_path = None
+
+
+class EventHandler:
+    def __init__(self, connection: QuicConnectionProtocol, global_dict: Dict):
+        self.connection = connection
+        self.global_dict = global_dict
+
+    def handle_client_indication(
+            self,
+            origin: str,
+            query: Dict[str, str]) -> None:
+        name = 'handle_client_indication'
+        if name in self.global_dict:
+            self.global_dict[name](self.connection, origin, query)
+
+    def handle_event(self, event: QuicEvent) -> None:
+        name = 'handle_event'
+        if name in self.global_dict:
+            self.global_dict[name](self.connection, event)
+
+
+class QuicTransportProtocol(QuicConnectionProtocol):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.streams = dict()
+        self.pending_events = []
+        self.client_indication_finished = False
+        self.client_indication_data = b''
+        self.handler = None
+
+    def quic_event_received(self, event: QuicEvent) -> None:
+        prefix = '!!'
+        logging.log(logging.INFO, 'QUIC event: %s' % type(event))
+        try:
+            if (not self.client_indication_finished and
+                    isinstance(event, StreamDataReceived) and
+                    event.stream_id == 2):
+                # client indication process
+                self.client_indication_data += event.data
```

Comment:
Could you add a size limit check here?  The limit we currently specify in the draft is 65,535 bytes.

Reply:
Addressed in https://github.com/web-platform-tests/wpt/pull/23705.